File: ObjectiveC SDK/CommonViewController.m
//
//  CommonViewController.m
//  ObjectiveC SDK
//
//  Created by Yuji on 2015/**/**.
//  Copyright (c) 2015年 Star Micronics. All rights reserved.
//

#import "CommonViewController.h"

@interface CommonViewController ()

@end

@implementation CommonViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (BOOL)getBlind {
    return self.navigationItem.hidesBackButton;
}

- (void)setBlind:(BOOL)blind {
    if (blind == YES) {
        self.navigationItem.hidesBackButton = YES;
        
        self.navigationItem.rightBarButtonItem.enabled = NO;
        
        _blindView            .hidden = NO;
        _activityIndicatorView.hidden = NO;
        
        [_activityIndicatorView startAnimating];
        
        [NSRunLoop.currentRunLoop runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];     // Update View
    }
    else {
        [_activityIndicatorView stopAnimating];
        
        _blindView            .hidden = YES;
        _activityIndicatorView.hidden = YES;
        
        self.navigationItem.rightBarButtonItem.enabled = YES;
        
        self.navigationItem.hidesBackButton = NO;
        
//      [NSRunLoop.currentRunLoop runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];     // Update View(No need)
    }
}

- (void)appendRefreshButton:(SEL)action {
    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemRefresh
                                                                          target:self
                                                                          action:action];
    
    self.navigationItem.rightBarButtonItem = item;
}

@end

File: Software/StarPRNT SDK/Swift SDK/Swift SDK/CommonViewController.swift
//
//  CommonViewController.swift
//  Swift SDK
//
//  Created by Yuji on 2015/**/**.
//  Copyright © 2015年 Star Micronics. All rights reserved.
//

import UIKit

class CommonViewController: UIViewController {
    @IBOutlet weak var blindView: UIView!
    
    @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    var blind: Bool {
        get {
            return self.navigationItem.hidesBackButton
        }
        
        set(blind) {
            if blind == true {
                self.navigationItem.hidesBackButton = true
                
                self.navigationItem.rightBarButtonItem?.enabled = false
                
                self.blindView            .hidden = false
                self.activityIndicatorView.hidden = false
                
                self.activityIndicatorView.startAnimating()
                
                NSRunLoop.currentRunLoop().runUntilDate(NSDate(timeIntervalSinceNow: 0.1))     // Update View
            }
            else {
                self.activityIndicatorView.stopAnimating()
                
                self.blindView            .hidden = true
                self.activityIndicatorView.hidden = true
                
                self.navigationItem.rightBarButtonItem?.enabled = true
                
                self.navigationItem.hidesBackButton = false
                
//              NSRunLoop.currentRunLoop().runUntilDate(NSDate(timeIntervalSinceNow: 0.1))     // Update View(No need)
            }
        }
    }
    
    func appendRefreshButton(action: Selector) {
        let item: UIBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Refresh, target: self, action: action)
        
        self.navigationItem.rightBarButtonItem = item
    }
}
